src/util/transfer.ts

```diff
@@ -0,0 +1,365 @@
+//line246
+// import { mergeMap, timeout, filter, map, first, skip } from 'rxjs/operators';
+import { environment } from '../environment/environment.js';
+import { networkState } from "@/state/networkState";
+import { NetworkStateUtils } from "@/state/utils/networkStateUtils";
+import { walletState } from "@/state/walletState";
+import { ChainUtils } from '@/util/chainUtils';
+import { WalletUtils } from '@/util/walletUtils'
+import { Helper } from "@/util/typeHelper";
+const config = require("@/../config/config.json");
+
+
+async function getAccInfo(address :string) :Promise<PublicAccount> {
+  // return await getPublicKey(recipientAddress, siriusStore.accountHttp);
+
+  let accountInfo = await WalletUtils.getAccInfo(address).then(accountinfo => accountinfo.publicAccount);
+  return accountInfo;
+}
+
+
+// export const getMosaicsAllAccounts = (appStore, siriusStore) => {
+//   const wallet = appStore.getWalletByName(appStore.state.currentLoggedInWallet.name);
+//   wallet.accounts.forEach((account) => {
+//     const mosaicService = new MosaicService(siriusStore.accountHttp, siriusStore.mosaicHttp, siriusStore.namespaceHttp);
+//     const address = Address.createFromRawAddress(account.address);
+//     mosaicService
+//       .mosaicsAmountViewFromAddress(address)
+//       .pipe(
+//           mergeMap((_) => _)
+//       )
+//       .subscribe(mosaic => {
+//         console.warn(mosaic)
+//         // account.mosaic.push({ id: mosaic.fullName(), amount: mosaic.relativeAmount(), divisibility: mosaic.mosaicInfo.divisibility });
+//       },
+//         err => console.error(err));
+//   });
+// };
+
+export const createTransaction = async (recipient :string, sendXPX :string, messageText :string, mosaicsSent :{amount: number ,id :string}[], mosaicDivisibility :number, walletPassword :string, senderAccName :string, cosigner :string, encryptedMsg :string) : Promise<announceAggregateBonded> => {
+  // verify password
+  // console.log('Pw after createTransaction: ' + walletPassword);
+  let verify = WalletUtils.verifyWalletPassword(walletState.currentLoggedInWallet.name, networkState.chainNetworkName, walletPassword)
+  /*let verify = appStore.verifyWalletPassword(appStore.state.currentLoggedInWallet.name, walletPassword); 
+ */
+  if (!verify) {
+    return verify;
+  }
+
+  const hash = networkState.currentNetworkProfile.generationHash
+  /* const add = fetch(siriusStore._buildAPIEndpointURL(siriusStore.state.selectedChainNode) + '/block/1').then((res) => res.json()).then((data) => { return data.meta.generationHash }); */
+
+
+  let networkType = networkState.currentNetworkProfile.network.type
+  const recipientAddress = recipient;
+
+  let xpxAmount = parseFloat(sendXPX) * Math.pow(10, 6);
+
+  let mosaics = [];
+  if (xpxAmount > 0) {
+    mosaics.push(new Mosaic(new MosaicId(environment.mosaicXpxInfo.id), UInt64.fromUint(Number(xpxAmount))));
```

Comment:
replace environment.mosaicXpxInfo.id to the 1 from the networkState, find the network currency

Reply:
networkState.currentNetworkProfile.network.currency.name 
 correct?